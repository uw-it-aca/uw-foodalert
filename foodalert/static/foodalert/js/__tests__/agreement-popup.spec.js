"use strict";
import Vue from 'vue';
import BootstrapVue from 'bootstrap-vue'
Vue.use(BootstrapVue);

import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome'
Vue.component('font-awesome-icon', FontAwesomeIcon);

import { library } from '@fortawesome/fontawesome-svg-core'
import {
    faUtensils,
    faChevronLeft,
    faCalendar,
    faClock,
    faMapMarkerAlt,
    faClipboardCheck,
    faBell,
} from '@fortawesome/free-solid-svg-icons'

library.add(faUtensils)
library.add(faChevronLeft)
library.add(faCalendar)
library.add(faClock)
library.add(faBell)
library.add(faMapMarkerAlt)
library.add(faClipboardCheck)

import {mount} from '@vue/test-utils';
import AgreePop from '../components/agreement-popup.vue';

describe('Agreement Popup', function () {
    describe('renders correctly', function () {
        test('textarea', function () {
            var wrapper = mount(AgreePop, {
                propsData: {
                    inputType: "number",
                    mainText: "Main TExt",
                    introText: "Intro Text",
                    infoText: "Info Text",
                    linkText: "Link Text",
                    linkLocation: "www.linklocation.com",
                    primaryText: "Primary Text",
                    secondaryText: "Secondary Text",
                    canBack: true,
                    primaryAction: jest.fn(),
                    secondaryAction: jest.fn(),
                    backAction: jest.fn(),
                }
            });
            expect(wrapper.element).toMatchSnapshot();
        });

        test('checkbox', function () {
            var wrapper = mount(AgreePop, {
                propsData: {
                    inputType: "checkbox",
                    checkboxOptions: ["Option 1", "Option 2"],
                    mainText: "Main TExt",
                    introText: "Intro Text",
                    infoText: "Info Text",
                    linkText: "Link Text",
                    linkLocation: "www.linklocation.com",
                    primaryText: "Primary Text",
                    secondaryText: "Secondary Text",
                    canBack: true,
                    primaryAction: jest.fn(),
                    secondaryAction: jest.fn(),
                    backAction: jest.fn(),
                }
            });
            expect(wrapper.element).toMatchSnapshot();
        })
    });
})
