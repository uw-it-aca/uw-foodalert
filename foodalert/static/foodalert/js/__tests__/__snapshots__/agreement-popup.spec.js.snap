// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Agreement Popup renders correctly checkbox 1`] = `
<div
  class="px-4"
>
  <a
    class="ml-0 mb-4"
  >
     Back 
  </a>
   
  <p>
    <br />
  </p>
   
  <p
    class="h5 text-center mt-4 mb-5 text-center font-weight-light"
  >
       Intro Text
  </p>
   
  <p
    class="h5 text-center mb-3"
  >
     Main TExt 
  </p>
   
  <fieldset
    class="b-form-group form-group"
    id="__BVID__6"
    role="group"
  >
    <!---->
    <div
      class=""
      role="group"
    >
      <!---->
       
      <div
        class="ml-5 custom-controls-stacked"
        id="__BVID__7"
        role="group"
        tabindex="-1"
      >
        <div
          class="custom-control custom-checkbox"
        >
          <input
            autocomplete="off"
            class="custom-control-input"
            id="__BVID__7__BV_check_0_opt_"
            name="__BVID__7"
            true-value="Option 1"
            type="checkbox"
            value="Option 1"
          />
          <label
            class="custom-control-label"
            for="__BVID__7__BV_check_0_opt_"
          >
            <span>
              Option 1
            </span>
          </label>
        </div>
        <div
          class="custom-control custom-checkbox"
        >
          <input
            autocomplete="off"
            class="custom-control-input"
            id="__BVID__7__BV_check_1_opt_"
            name="__BVID__7"
            true-value="Option 2"
            type="checkbox"
            value="Option 2"
          />
          <label
            class="custom-control-label"
            for="__BVID__7__BV_check_1_opt_"
          >
            <span>
              Option 2
            </span>
          </label>
        </div>
      </div>
      <!---->
      <!---->
      <!---->
    </div>
  </fieldset>
   
  <div
    class="d-flex justify-content-center"
  >
    <button
      class="btn px-5 btn-secondary btn-lg"
      type="button"
    >
      
            Secondary Text
        
    </button>
     
    <button
      class="btn btn-primary btn-lg disabled px-5 ml-2"
      disabled="disabled"
      type="button"
    >
      
            Primary Text
        
    </button>
  </div>
   
  <br />
   
  <p
    class="d-block text-center w-100 text-center mt-4 mb-3 font-italic"
  >
     Info Text 
  </p>
   
  <a
    class="d-block text-center mb-3"
    href="www.linklocation.com"
  >
     Link Text 
  </a>
</div>
`;

exports[`Agreement Popup renders correctly textarea 1`] = `
<div
  class="px-4"
>
  <a
    class="ml-0 mb-4"
  >
     Back 
  </a>
   
  <p>
    <br />
  </p>
   
  <p
    class="h5 text-center mt-4 mb-5 text-center font-weight-light"
  >
       Intro Text
  </p>
   
  <p
    class="h5 text-center mb-3"
  >
     Main TExt 
  </p>
   
  <fieldset
    class="b-form-group form-group"
    id="__BVID__2"
    role="group"
  >
    <!---->
    <div
      class=""
      role="group"
    >
      <input
        class="text-center mb-3 border-0 form-control"
        id="__BVID__3"
        placeholder="40569503"
        type="number"
      />
       
      <!---->
      <!---->
      <!---->
      <!---->
    </div>
  </fieldset>
   
  <div
    class="d-flex justify-content-center"
  >
    <button
      class="btn px-5 btn-secondary btn-lg"
      type="button"
    >
      
            Secondary Text
        
    </button>
     
    <button
      class="btn btn-primary btn-lg disabled px-5 ml-2"
      disabled="disabled"
      type="button"
    >
      
            Primary Text
        
    </button>
  </div>
   
  <br />
   
  <p
    class="d-block text-center w-100 text-center mt-4 mb-3 font-italic"
  >
     Info Text 
  </p>
   
  <a
    class="d-block text-center mb-3"
    href="www.linklocation.com"
  >
     Link Text 
  </a>
</div>
`;
